### PR TITLE
Gaj/feature/items component

### PR DESCRIPTION
**Pull request by kelsiebesingeryeh**
- [x ] Feature
- [ ] Refactor
- [ ] Bug Fix 
- [ ] Clean Up

Added Item and Cosmetic components in order for a user to view a specific product type's products - i.e Vegan => Blush => All blushes that match. Styled the cosmetics card.

Added Item and Cosmetic components in order for a user to view a specific product type's products - i.e Vegan => Blush => All blushes that match. Styled the cosmetics card.

App, Item and Cosmetic Component files. Added Item component as the parent component and Cosmetic component as the child of Item. Linked the Item component inside of App component.

npm start and go to localhost:3000. Navigate to any of the three categories - vegan, allergenFriendly, or Eco and click on the various product types.

#### Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x ] My code follows the style guidelines of this project
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -127,6 +128,238 @@ class App extends Component {
...
+            match.params.type === "nail_polish"
+          ) {
+            return <Items
+                      data={this.state.allergenFriendly}
+                      type='nail_polish'
+                    />;
+          }
+        }}
+        />
+
```
Comment (conconartist) [inline src/components/App/App.js:362]: I definitely think we can refactor this-- glad it's working for now!

File: src/components/Cosmetic/Cosmetic.css [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+   margin-bottom: 1rem;
+}
+
+.cosmeticDetail {
+    margin: 0;
+}
+
+.tags {
+    border: 1px solid black;
+    padding: .5rem;
```
Comment (conconartist) [inline src/components/Cosmetic/Cosmetic.css:39]: Nice job with the styling!

File: src/components/Cosmetic/Cosmetic.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        <img src={img} alt={name} className="cosmeticCardImage"></img>
+        <p className="cosmeticDetail">
+          <strong>{brand}</strong>
+        </p>
+        <p className="cosmeticDetail">{name}</p>
+        <p className="tags">{styledTags}</p>
+      </article>
+  );
+}
+
```
Comment (conconartist) [inline src/components/Cosmetic/Cosmetic.js:19]: Good work on this-- excited we're able to get our products to display.

File: src/components/Type/Type.js [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
+const Type = ({img, title, category}) => {
     return (
-      <article className="productType" style={{backgroundImage:`url(${img})`}}>
-        <h3>{title}</h3>
-      </article>
+      <Link to={`/${category}/${title}`}>
+        <article className="productType" style={{backgroundImage:`url(${img})`}}>
+          <h3>{title}</h3>
+        </article>
+      </Link>
```
Comment (conconartist) [inline src/components/Type/Type.js:11]: Thanks for working that in!

**Review by conconartist — APPROVED** [review]

File: src/components/App/App.js [hunk 5]
```diff
@@ -127,6 +128,238 @@ class App extends Component {
             }}
           />
 
+      <Route
+        exact
```
Comment (gaj23) [inline src/components/App/App.js:132]: I'm glad this works for us. We can refactor at a later time.

File: src/components/Cosmetic/Cosmetic.css [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+.cosmeticCard {
+        box-sizing: border-box;
+        display: flex;
+        flex-direction: column;
+        justify-content: flex-start;
+        border: 1px solid black;
+        border-radius: 8px;
+        align-items: center;
+        background: white;
```
Comment (gaj23) [inline src/components/Cosmetic/Cosmetic.css:18]: We can add our off-white color layer to remain consistent to our color scheme.

File: src/components/Cosmetic/Cosmetic.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import './Cosmetic.css';
+import { Link } from "react-router-dom";
+
+const Cosmetic = ({name, brand, img, tags, productType, id}) => {
+  const styledTags = tags.join(', ');
+
+  return (
```
Comment (gaj23) [inline src/components/Cosmetic/Cosmetic.js:8]: This is great! Thank you for this work.

File: src/components/Items/Items.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import Cosmetic from '../Cosmetic/Cosmetic';
+import './Items.css';
+
+const Items = ({ data, type }) => {
```
Comment (gaj23) [inline src/components/Items/Items.js:5]: I'm proud of our solving problem here.

File: src/components/Type/Type.js [hunk 9]
```diff
@@ -1,11 +1,14 @@
...
 import './Type.css';
+import {Link} from 'react-router-dom';
 
-const Type = ({img, title}) => {
+const Type = ({img, title, category}) => {
     return (
-      <article className="productType" style={{backgroundImage:`url(${img})`}}>
-        <h3>{title}</h3>
-      </article>
+      <Link to={`/${category}/${title}`}>
```
Comment (gaj23) [inline src/components/Type/Type.js:7]: This interpolation has helped me see how we might be able to try up our category components.

**Review by gaj23 — APPROVED** [review]
Thank you for this work. Iteration 3 is almost done.

**Pull request merged**